File: app/schemas/member.py
"""
API schemas for the member management endpoints (POST/GET /v1/members).
"""

from __future__ import annotations

import uuid
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class MemberCreate(BaseModel):
    email: str = Field(min_length=1, max_length=320)
    name: str = Field(min_length=1, max_length=255)


class MemberCreated(BaseModel):
    user_id: uuid.UUID
    membership_id: uuid.UUID
    status: Literal["pending"] = "pending"


class MemberRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    user_id: uuid.UUID
    membership_id: uuid.UUID
    email: str
    name: str | None
    role: str
    status: Literal["pending", "accepted"]
    created_at: datetime


class MemberListResponse(BaseModel):
    items: list[MemberRead]
    next_cursor: str | None = None

File: packages/api-client/src/contracts/member.ts
/** Member management contracts — mirrors app/schemas/member.py */

export interface MemberCreate {
  email: string;
  name: string;
}

export interface MemberCreated {
  user_id: string;
  membership_id: string;
  status: "pending";
}

export interface MemberRead {
  user_id: string;
  membership_id: string;
  email: string;
  name: string | null;
  role: string;
  status: "pending" | "accepted";
  created_at: string;
}

export interface MemberListResponse {
  items: MemberRead[];
  next_cursor: string | null;
}
